### PR TITLE
Update to elasticlunr-rs 3.0.0

### PR DESCRIPTION
**Pull request by mattico**
I'm planning to release elasticlunr-rs 3.0.0 soon. The one noticeable change for you is the new `IndexBuilder::add_field_with_tokenizer` function.

I wanted to give a chance for feedback before I release the final version.

### CODE REVIEW

File: components/search/src/lib.rs [hunk 1]
```diff
@@ -55,34 +52,6 @@ fn path_tokenizer(text: &str) -> Vec<String> {
...
 }
 
-fn build_tokenizers(search_config: &Search, language: Language) -> Vec<TokenizerFn> {
-    let text_tokenizer = match language {
-        #[cfg(feature = "indexing-zh")]
-        Language::Chinese => pipeline::tokenize_chinese,
-        #[cfg(feature = "indexing-ja")]
-        Language::Japanese => pipeline::tokenize_japanese,
-        _ => pipeline::tokenize,
-    };
```
Comment (Keats) [inline components/search/src/lib.rs:65]: Additional languages gated by features work automatically now?

**Review by Keats — COMMENTED** [review]
lgtm

File: components/search/src/lib.rs (hunk 1 above)
Comment (mattico) [reply components/search/src/lib.rs:65]: Tokenization is now included in the language trait.

Comment (Keats) [reply components/search/src/lib.rs:65]: Looks good then, I'll merge it when 3.0 is released

**Review by Keats — APPROVED** [review]

**Pull request merged**